### PR TITLE
Unit42 playbook viewer

### PR DESCRIPTION
**Pull request by teizenman**
## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/28412

## Description
- Added the processing of the STIX Report objects

## Screenshots
The update for the related indicator widget
image

### CODE REVIEW

**Review by bakatzir — COMMENTED** [review]
I guess you were planning to add UT for the reports IOC as well, right?

File: Packs/FeedUnit42/Integrations/FeedUnit42/FeedUnit42.py [hunk 1]
```diff
@@ -104,61 +96,231 @@ def parse_indicators(objects: list, feed_tags: list = [], tlp_color: Optional[st
...
-                                if 'url' in item or 'external_id' in item:
-                                    feedrelatedindicators_obj = {
-                                        'type': 'MITRE ATT&CK',
-                                        'value': item.get('external_id'),
-                                        'description': item.get('url')
-                                    }
-                                    indicator['fields'][field_type].extend([feedrelatedindicators_obj])
+        for relation in matched_relationships.get(indicator_id, []):
+            relation_object = id_to_object.get(relation)
+            if not relation_object:
```
Comment (bakatzir) [inline Packs/FeedUnit42/Integrations/FeedUnit42/FeedUnit42.py:114]: please add a comment for the dev in the future to come

**Review by bakatzir — COMMENTED** [review]
Also, please re consider testing more logic in the test_module by parsing the stix objects and the relationships as well.

**Comment by teizenman** [comment]
> Also, please re consider testing more logic in the test_module by parsing the stix objects and the relationships as well.

Since we are fetching a small number of objects, there is no guarantee that we fetch the right ones for the parsing.
There are UTs to cover this issue,  and the test module tests the connection to the feed itself.

File: Packs/CommonScripts/Scripts/FeedRelatedIndicatorsWidget/FeedRelatedIndicatorsWidget.py [hunk 2]
```diff
@@ -2,33 +2,67 @@
...
+        Dict. Related indicator object.
+    """
+    indicator_object = {
+        'Value': value,
+        'Type': type_
+    }
+
+    descriptions = []
+
+    for desc in description.split(','):
```
Comment (bakatzir) [inline Packs/CommonScripts/Scripts/FeedRelatedIndicatorsWidget/FeedRelatedIndicatorsWidget.py:23]: what will happen if there is only one description?
I think we can use argToList here.

Comment (bakatzir) [reply Packs/CommonScripts/Scripts/FeedRelatedIndicatorsWidget/FeedRelatedIndicatorsWidget.py:23]: and add a test case for single and multiple descriptions.

File: Packs/FeedUnit42/Integrations/FeedUnit42/FeedUnit42.py [hunk 3]
```diff
@@ -104,61 +96,264 @@ def parse_indicators(objects: list, feed_tags: list = [], tlp_color: Optional[st
...
+      objects: a list of objects.
+
+    Returns:
+        List. Objects of type report.
+    """
+    main_report_objects = []
+    sub_report_objects = []
+
+    for obj in objects:
+        is_main = False
```
Comment (bakatzir) [inline Packs/FeedUnit42/Integrations/FeedUnit42/FeedUnit42.py:173]: can we rename to `is_main_report`?

File: Packs/FeedUnit42/Integrations/FeedUnit42/FeedUnit42.py [hunk 4]
```diff
@@ -181,14 +387,33 @@ def fetch_indicators(client: Client, feed_tags: list = [], tlp_color: Optional[s
     Returns:
         Indicators.
```
Comment (bakatzir) [inline Packs/FeedUnit42/Integrations/FeedUnit42/FeedUnit42.py:388]: let's update the return, what types of indicators

**Review by bakatzir — COMMENTED** [review]
a few last comments, looks really good.
please attach a printscreen of the widget, or dm me a short demo of the fetch indicators and IOC/Report update in the indicators tab -> widget update

just to make sure that it all plays well.

**Review by bakatzir — APPROVED** [review]
I'm good with the changes. make sure you pass LGTM and the build. and let's wait a few hours to see if Anar/Lichtman would wish a short demo.

**Comment by teizenman** [comment]
@roysagi 
Can you please go over the RN please?

**Comment by teizenman** [comment]
* MT support has been verified - it works for the main server and for a tenant without a leading `/`.
* Multiprocessing removed since it takes the same amount of time without it.
* Malware description has been updated with a fall back to labels.

**Review by bakatzir — COMMENTED** [review]
looks good. only fixes to the `stix report` layout are needed:
add the type of stix report to the layout, just like in Domain.
rm Actions
split report details to report details and indicator details

**Comment by teizenman** [comment]
We have worked with Olga, and this is the new layout sections:

image

**Review by bakatzir — APPROVED** [review]

**Pull request merged**